projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -4,222 +4,263 @@ import apiFetch from '@wordpress/api-fetch';
 			}
-			setTimeout( () => setAttributes( { content: content } ), 50 * tokens.length );
-		} )
-		.catch( () => {
-			setErrorMessage(
-				__(
-					'Whoops, we have encountered an error. AI is like really, really hard and this is an experimental feature. Please try again later.',
-					'jetpack'
-				)
-			);
-			setLoadingCompletion( false );
-		} );
-}
-
-function createPrompt( title = '', content = '', categoryNames = '' ) {
-	// If title is not added, we will only complete.
-	if ( ! title ) {
-		return content;
+			setTimeout( () => setContent( html ), 50 * tokens.length );
+		} else {
+			setContent( html );
+		}
+		setContentSet( true );
 	}
 
-	// Title, content and categories
-	if ( content && categoryNames.length ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, comma-seperated category names and the last 240 characters of content. */
-			__( "This is a post titled '%1$s', published in categories '%2$s':\n\n … %3$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			categoryNames,
-			content
-		);
-	}
+	return (
+		<>
+			<div className="content">
+				<RawHTML>{ content }</RawHTML>
+			</div>
+		</>
+	);
+}
 
-	// No content, only title and categories
-	if ( categoryNames.length ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and comma-seperated category names. */
-			__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			categoryNames
-		);
-	}
+export default function Edit( { attributes, setAttributes, clientId } ) {
+	const [ loadingCompletion, setLoadingCompletion ] = useState( false );
+	const [ loadingCategories, setLoadingCategories ] = useState( false );
+	const [ showAnimation, setShowAnimation ] = useState( false );
+	const [ needsMoreCharacters, setNeedsMoreCharacters ] = useState( 0 );
+	const [ showRetry, setShowRetry ] = useState( false );
+	const [ completionFinished, setCompletionFinished ] = useState( !! attributes.content );
+	const [ errorMessage, setErrorMessage ] = useState( false );
 
-	// Title and content
-	if ( content ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and the last 240 characters of content. */
-			__( "This is a post titled '%1$s':\n\n…%2$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			content
-		);
-	}
+	// Let's grab post data so that we can do something smart.
+	const currentPostTitle = useSelect(
+		select => select( 'core/editor' ).getEditedPostAttribute( 'title' ),
+		[]
+	);
 
-	return sprintf(
-		/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
-		__( 'Write content of a post titled "%s"', 'jetpack' ),
-		title
+	let loading = false;
```

Comment:
`isLoadingCategories` or `isLoadingTaxonomies` might be a better name.  I had to dive into the code to find out what the loading was doing and if it needed a react hook 😆.

Reply:
`isLoadingTaxonomiesForSettingLater` seems a bit much.

Reply:
Done in e99b74e928f22c07caee547614267b38b6d10906